### PR TITLE
Rename uses south attribute

### LINKED ISSUE #161: Rename "uses_south" setting
One of our applications uses Django 1.8 now, which has native support of migrations, we don't use `South` there. However, we still have to use `uses_south = True` setting in WLProxy deploy hooks, because otherwise migrations won't be run on deploy (https://github.com/yola/yodeploy/blob/23ca59d1358c8f0bc8b97423d518b3656d7c91bd/yodeploy/hooks/django.py#L112)

I think we can think how to rename this setting to something more neutral, not related to South.

### PR DESCRIPTION
**Pull request by ekoval**
`uses_south` flag in DjangoApp hook is renamed to `has_migrations` to be compliant with Django 1.8+, which doesn't need South for migrations.

Plus some following Boy Scout rule :)

Closes #161

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def parse_requirements():
     author='Stefano Rivera',
     author_email='stefano.rivera@yola.com',
     url='https://github.com/yola/yodeploy',
-    version='0.5.1',
+    version=yodeploy.__version__,
```
Comment (snitch) [inline setup.py]: F821 undefined name 'yodeploy'

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by adrianmoisey** [comment]
This currently a backwards-incompatible change, either it needs to be made backwards compatible or it should get a major version bump.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
 Change Log
 ==========
 
+0.5.2
+------
+
+* `uses_south` flag in DjangoApp hook is renamed to `has_migrations` to be
+  compliant with Django 1.8+, which doesn't need South for migrations.
```
Comment (stefanor) [inline CHANGELOG.md]: Backwards-incompatible changes should be accompanied by a minor version number bump (major if we were post 1.0)

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by RayeN** [comment]
:+1: Looks good to me if @stefanor doesn't have any objections.

**Pull request merged**